pkg/xds/sync/ingress_proxy_builder.go

```diff
@@ -46,11 +48,15 @@ func (p *IngressProxyBuilder) build(key core_model.ResourceKey, streamId int64)
 		return nil, err
 	}
 
+	metadata := p.MetadataTracker.Metadata(streamId)
+	if metadata == nil {
+		return nil, errors.New("Metadata is empty for stream")
```

Comment:
Nit: error strings should not be capitalized.

https://github.com/golang/go/wiki/CodeReviewComments#error-strings